src/Raven.Server/Documents/Replication/DocumentInfoHelper.cs

```diff
@@ -26,7 +26,17 @@ public unsafe LazyStringValue GetDocumentId(Slice key)
             _tmpLazyStringInstance.Renew(null, key.Content.Ptr, sepIdx, _context);
             return _tmpLazyStringInstance;
         }
-        
+
+        public unsafe LazyStringValue GetDocumentId(LazyStringValue key)
+        {
+            var index = key.IndexOf((char)SpecialChars.RecordSeparator, StringComparison.OrdinalIgnoreCase);
+            if (index == -1)
+                return null;
+
+            _tmpLazyStringInstance = _context.GetLazyString(key.Buffer, index);
```

Comment:
Why do you store that in the temporary variable?

Reply:
done